From 7c742f0cb966c214358c0e03890f5ce66e117063 Mon Sep 17 00:00:00 2001
From: Randall Rushing <randall.rushing@gmail.com>
Date: Sun, 10 Jan 2016 12:09:58 -0800
Subject: [PATCH 49/54] Themes: Allow volume slider and thumb to work with
 custom images.

Credit Bryan Owens for giving guidance on what was needed.

Change-Id: I862008d1b433666333a4c77904b44a05e70472a2
---
 packages/SystemUI/src/com/android/systemui/volume/VolumeDialog.java | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/packages/SystemUI/src/com/android/systemui/volume/VolumeDialog.java b/packages/SystemUI/src/com/android/systemui/volume/VolumeDialog.java
index 317834605a7..10be65f9dfa 100644
--- a/packages/SystemUI/src/com/android/systemui/volume/VolumeDialog.java
+++ b/packages/SystemUI/src/com/android/systemui/volume/VolumeDialog.java
@@ -29,6 +29,7 @@ import android.content.res.Configuration;
 import android.content.res.Resources;
 import android.graphics.Color;
 import android.graphics.PixelFormat;
+import android.graphics.PorterDuff;
 import android.graphics.Rect;
 import android.graphics.drawable.AnimatedVectorDrawable;
 import android.graphics.drawable.ColorDrawable;
@@ -387,6 +388,8 @@ public class VolumeDialog implements TunerService.Tunable {
         row.header.setId(20 * row.stream);
         mSpTexts.add(row.header);
         row.slider = (SeekBar) row.view.findViewById(R.id.volume_row_slider);
+        row.slider.setProgressTintMode(PorterDuff.Mode.SRC_ATOP);
+        row.slider.setThumbTintMode(PorterDuff.Mode.SRC_ATOP);
         row.slider.setOnSeekBarChangeListener(new VolumeSeekBarChangeListener(row));
         row.anim = null;
         row.cachedShowHeaders = VolumePrefs.DEFAULT_SHOW_HEADERS;
-- 
2.11.0

